### PR TITLE
Fix license header checking during the build process

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

**Comment by johann-beleites-sonarsource** [comment]
Please don't squash

File: sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/surefire/data/package-info.java [hunk 1]
```diff
@@ -17,5 +17,4 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-@javax.annotation.ParametersAreNonnullByDefault
```
Comment (margarita-nedzelska-sonarsource) [inline sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/surefire/data/package-info.java:20]: I think we still need this

File: sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/surefire/package-info.java [hunk 2]
```diff
@@ -17,5 +17,4 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-@javax.annotation.ParametersAreNonnullByDefault
```
Comment (margarita-nedzelska-sonarsource) [inline sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/surefire/package-info.java:20]: I think we still need this

File: sonar-kotlin-plugin/src/main/java/org/sonarsource/slang/externalreport/package-info.java [hunk 3]
```diff
@@ -17,7 +17,6 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-@ParametersAreNonnullByDefault
```
Comment (margarita-nedzelska-sonarsource) [inline sonar-kotlin-plugin/src/main/java/org/sonarsource/slang/externalreport/package-info.java:20]: I think we still need this

File: sonar-kotlin-plugin/build.gradle.kts [hunk 4]
```diff
@@ -1,13 +1,43 @@
...
     kotlin("jvm")
+    id("com.diffplug.spotless") version "5.15.0"
+}
+
+configure<com.diffplug.gradle.spotless.SpotlessExtension> {
+
+    lineEndings = com.diffplug.spotless.LineEnding.UNIX
+
+    kotlin {
+        // by default the target is every '.kt' and '.kts` file in the java sourcesets
```
Comment (margarita-nedzelska-sonarsource) [inline sonar-kotlin-plugin/build.gradle.kts:18]: maybe this comment is not relevant

**Review by margarita-nedzelska-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**